cpp/inc/bond/core/nullable.h

```diff
@@ -482,32 +474,37 @@ class nullable<T, Allocator, false>
 #endif
 
 private:
-    template<typename AllocatorT>
-    void destroy(AllocatorT& alloc)
-    {
 #ifndef BOND_NO_CXX11_ALLOCATOR
-        std::allocator_traits<AllocatorT>::destroy(alloc, _value);
+    typedef typename std::allocator_traits<allocator_type>::
+        template rebind_alloc<value_type> rebind_alloc;
+
+    typedef typename std::allocator_traits<rebind_alloc>::pointer real_pointer;
 #else
-        _value->~T();
+    typedef typename allocator_type::template rebind<value_type>::other rebind_alloc;
+
+    typedef pointer real_pointer;
```

Comment:
Above we get `real_pointer` from the traits of `rebind_alloc`, but here we get it from ourselves (where it is `T*`) instead of from `rebind_alloc`. Intentional difference?